libbeat/docs/monitoring/configuring.asciidoc

```diff
@@ -0,0 +1,178 @@
+    username: elastic
+    password: changeme
+--------------------
+
+Many of the configuration options for `xpack.monitoring` match the
+`output.elasticsearch` options. If you configured {es} output and you want to
+use the same {es} cluster and credentials to store your monitoring metrics, you
+can specify the following minimal configuration options:
+
+[source, yml]
+--------------------
+xpack.monitoring:
+  enabled: true
+  elasticsearch:
+--------------------
+
+--
+
+. {kibana-ref}/monitoring-xpack-kibana.html[Configure monitoring in {kib}].
+
+. To verify your monitoring configuration, point your web browser at your {kib}
+host, and select Monitoring from the side navigation. Metrics reported from
+{beatname_uc} should be visible in the Beats section. When {security} is enabled,
+to view the monitoring dashboards you must log in to {kib} as a user who has the
+`kibana_user` and `monitoring_user` roles.
+
+[float]
+=== Configuration options
+
+You can specify the following options in the `xpack.monitoring` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+==== `enabled`
+
+The `enabled` config is a boolean setting to enable or disable {monitoring}.
+If set to `true`, monitoring is enabled.
+
+The default value is `false`.
+
+[float]
+==== `elasticsearch`
+
+The {es} instances that you want to ship your {beatname_uc} metrics to. This
+configuration option contains the following fields:
+
+[float]
+===== `bulk_max_size`
+
+The maximum number of metrics to bulk in a single {es} bulk API index request.
+The default is `50`. For more information, see <<elasticsearch-output>>.
+
+////
+[float]
+===== `buffer_size`
+
+TBD. The default value is `50`.
+////
+[float]
+===== `compression_level`
+
+The gzip compression level. Setting this value to `0` disables compression. The
+compression level must be in the range of `1` (best speed) to `9` (best
+compression). The default value is `0`. Increasing the compression level
+reduces the network usage but increases the CPU usage.
+
+[float]
+===== `headers`
+
+Custom HTTP headers to add to each request. For more information, see
+<<elasticsearch-output>>.
+
+[float]
+===== `hosts`
+
+The list of {es} nodes to connect to. Monitoring metrics are distributed to
+these nodes in round robin order. For more information, see
+<<elasticsearch-output>>.
+
+[float]
+===== `max_retries`
+
+The number of times to retry sending the monitoring metrics after a failure.
+After the specified number of retries, the metrics are typically dropped. The
+default value is `3`. For more information, see <<elasticsearch-output>>.
+
+[float]
+===== `parameters`
+
+Dictionary of HTTP parameters to pass within the url with index operations.
+
+[float]
+===== `password`
+
+The password that {beatname_uc} uses to authenticate with the {es} instances for
+shipping monitoring data.
+////
+[float]
+===== `period`
+
+TBD. The default value is 60 * time.Second.
+////
```

Comment:
Monitoring requires the beat to push/publish metrics every N seconds to elasticsearch. The `period` configures the time interval a new snapshot of beats metrics is generated and scheduled to be published. If output is unresponsive for some long time and queues are full, a snapshot might get missing.